evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -152,8 +152,7 @@ class SelectByType(ColumnSelector):
     def __init__(self, column_types=None, random_seed=0, **kwargs):
         parameters = {"column_types": column_types}
         parameters.update(kwargs)
-        Transformer.__init__(
```

Comment:
This was causing parameters to disappear. Still slightly unsure of the exact mechanism of how the parameters disappear but my hypothesis is that the `parameters` field in `SelectByType` you can see here is different from the `parameters` field created by this init logic.

Reply:
It's not that they're disappearing, it's that they're put in the `parameters` dictionary because that's that the `Transformer` init does. 

```
'Numeric Pipeline - Select Columns By Type Transformer': {'columns': None, 'parameters': {'column_types': ['numeric']}, 'component_obj': None}
```

I agree with the redesign. `ColumnSelectors` are meant to take a list of columns so we need to either need to rename `column_types` to `columns` or not make this component a subclass. What we here doing before of trying to skip the parent init and get to the grandparent init is probably not the right thing to do.

Interesting find!